synapse/push/mailer.py

```diff
@@ -153,6 +218,33 @@ def send_password_reset_mail(self, email_address, token, client_secret, sid):
             template_vars,
         )
 
+    @defer.inlineCallbacks
+    def send_registration_mail(self, email_address, token, client_secret, sid):
+        """Send an email with a registration confirmation link to a user
+
+        Args:
+            email_address (str): Email address we're sending the registration
+                link to
+            token (str): Unique token generated by the server to verify
+                the email was received
+            client_secret (str): Unique token generated by the client to
+                group together multiple email sending attempts
+            sid (str): The generated session ID
+        """
+        link = (
+            self.hs.config.public_baseurl
+            + "_matrix/client/unstable/registration/email/submit_token"
```

Comment:
we shouldn't be using /unstable/ for this (nor should we for password_reset, for that matter; we could fix that at the same time). Why don't we just use /r0/ to match the other endpoints?

Reply:
This was @erikjohnston's suggestion, and I think the reason was that this isn't a specced endpoint. It's simply left up to the implementation to decide.

But if we don't have a policy on unspecced things being under `/r0/` and/or want to reserve `/unstable/` for experimental stuff then we probably should move it.

Reply:
Actually, we need to add another endpoint for just adding an email to your account. So this function should be split up into multiple servlets.